javascript/test/config-file-tests.js

```diff
@@ -0,0 +1,165 @@
+            assert.strictEqual(minesweeperProcess.status, 2, "Wrong exit code for invalid board configuration file extension")
+        });
+
+        it('Minesweeper exit because of invalid characters (,)', function () {
+            // Creates the simple.cfg file
+            const lines = ['..*', '.,.', '...'];
+            fs.writeFileSync(tempDir + "simple.cfg", lines.join('\n') + '\n')
+
+            minesweeperProcess = spawnSync('node', [minesweeperHome + "minesweeper.js", tempDir + "simple.cfg"], { timeout: 3000 })
+            assert.strictEqual(minesweeperProcess.status, 2, "Wrong exit code for invalid (wrong characters) board configuration file")
+        });
+
+        it('Minesweeper exit because of new line character in last line missing', function () {
+            // Creates the simple.cfg file
+            const lines = ['..*', '...', '...'];
+            fs.writeFileSync(tempDir + "simple.cfg", lines.join('\n'))
+
+            minesweeperProcess = spawnSync('node', [minesweeperHome + "minesweeper.js", tempDir + "simple.cfg"], { timeout: 3000 })
+            assert.strictEqual(minesweeperProcess.status, 2, "Wrong exit code for invalid (new line missing) board configuration file")
+        });
+
+        it('Minesweeper exit because of only mines', function () {
+            // Creates the simple.cfg file
+            const lines = ['***', '***', '***'];
+            fs.writeFileSync(tempDir + "simple.cfg", lines.join('\n') + '\n')
+
+            minesweeperProcess = spawnSync('node', [minesweeperHome + "minesweeper.js", tempDir + "simple.cfg"], { timeout: 3000 })
+            assert.strictEqual(minesweeperProcess.status, 2, "Wrong exit code for invalid (only mines) board configuration file")
+        });
+
+        it('Minesweeper exit because of single square', function () {
+            // Creates the simple.cfg file
+            const lines = ['.'];
+            fs.writeFileSync(tempDir + "simple.cfg", lines.join('\n') + '\n')
+
+            minesweeperProcess = spawnSync('node', [minesweeperHome + "minesweeper.js", tempDir + "simple.cfg"], { timeout: 3000 })
+            assert.strictEqual(minesweeperProcess.status, 2, "Wrong exit code for invalid (single square) board configuration file")
+        });
+
+        it('Minesweeper exit because board is no rectangle', function () {
+            // Creates the simple.cfg file
+            const lines = ['..*', '..'];
+            fs.writeFileSync(tempDir + "simple.cfg", lines.join('\n') + '\n')
+
+            minesweeperProcess = spawnSync('node', [minesweeperHome + "minesweeper.js", tempDir + "simple.cfg"], { timeout: 3000 })
+            assert.strictEqual(minesweeperProcess.status, 2, "Wrong exit code for invalid (no rectangle) board configuration file")
+        });
+
+        it('Minesweeper exit because more than 20 columns', function () {
+            // Creates the simple.cfg file
+            const lines = ['....................*', '.....................'];
```

Comment:
Maybe something like `'.'.repeat(20+1)` is easier to read/understand here

[1] https://stackoverflow.com/questions/1877475/repeat-character-n-times